DOCS-2632 Monitoring HTTP Traffic Redirects to HTTPS Guide (#11945)
DataDog/documentation: content/en/synthetics/guide/monitor-https-redirections.md

---
title: Monitor your HTTP requests are redirected into HTTPS
kind: guide
further_reading:
    - link: '/synthetics/http_tests'
      tag: 'Documentation'
      text: 'Create a HTTP Test'

---

Monitoring that your HTTP traffic is redirected into HTTPS is key to ensure the encryption of your users connections with your API endpoints and your application.

## Monitor your HTTPS redirections

> Depending on your setup, you can find information about the redirection either in the response header with the presence of the `location` header or in the response body from your HTTP request (presence of `"https:"===window.location.protocol`).

Reviewer: ```suggestion

Depending on your setup, you can identify the redirect to HTTPS in the generated **Response Preview** tab under Headers as `location` or in the **Body** as `"https:"===window.location.protocol`.
```

> To start monitoring the redirection of your HTTP traffic into HTTPS:

Reviewer: ```suggestion
To monitor the redirection of your HTTP traffic into HTTPS:
```

> 1. Create an HTTP.
> 2. Click on the **Test Url** button and look for redirection information in the response preview.
> 3. Make an assertion on the presence of the redirection element.

Reviewer: ```suggestion

1. Create an HTTP test and [define the request][1].
2. Click **Test URL **. The response preview generates a **Request Preview** and **Response Preview**.
3. Click **+ New Assertion** or a `location` header to define an assertion on the redirection element.
```

<!-- Screenshot 1 et 2-->

> You will then be alerted if your HTTP traffic is not correctly redirected into HTTPS.

Reviewer: ```suggestion
Datadog alerts you if your HTTP traffic does not correctly redirected into HTTPS.
```

## Further Reading

> {{< partial name="whats-next/whats-next.html" >}}

Reviewer: ```suggestion
{{< partial name="whats-next/whats-next.html" >}}

[1]: /getting_started/synthetics/api_test/#define-request

```
